## Section 4: Genome Analysis

The process of genome annotation aims to assign function to DNA sequence.  In the simplest terms, it involves predicting the presence of functional RNA and protein coding genes.  Subsequently, these assignments are used to determine what the purpose the resulting products have inside cells.  It is not a trivial task and can be extremely time consuming.  It is also fraught with pitfalls and opportunities to do yourself a great disservice by confounding your ability to perform worthwhile downstream analysis.

We'll start the secion with a 5 minutes presentation from each person on what they have learned about their genome from the RAST annoation.  We'll then proceed to predicting some features of an assembled genome dataset.  We'll do this using some web-based tools and then I'll show you how to use linux bases software to do the same.

The following websites will come in handy for this:

rRNA prediction
http://www.cbs.dtu.dk/services/RNAmmer/
tRNA prediction
http://lowelab.ucsc.edu/tRNAscan-SE/
Protein prediction
http://linux1.softberry.com/berry.phtml?topic=fgenesb&group=programs&subgroup=gfindb
Annotation software
http://www.sanger.ac.uk/Software/Artemis/
Database Blast seraches
http://blast.ncbi.nlm.nih.gov/Blast.cgi
Enzyme descriptions
http://www.ebi.ac.uk/embl/

### RAST Annotation


### Predicting Proteins and Extracting ORFs



### Finding rRNA and tRNA genes



### Pairwise Blast -- Genome-to-Genome Comparison


### KEGG Annotation -- and KEGG Mapping




### Assessing Genome Completeness


## Metagenome Gene Frequencies

### Proportion of Arachae to Bacteria

### What proportion of a population carries a particular Gene ?
